lua/tests/infoview/run_spec.lua

```diff
@@ -78,7 +89,7 @@ describe('infoview', function()
       assert.is_false(vim.api.nvim_win_is_valid(infoview_info.window))
       assert.is.equal(num_wins - 1, get_num_wins())
       assert.is.equal(win, vim.api.nvim_get_current_win())
-      assert.is_falsy(infoview.is_open())
+      assert.is_not.open_state()
```

Comment:
This seems less readable than it was before to me.

Reply:
All this test did before was check that the internal state was updated (whether it was physically closed is done above). From that perspective, this would seem more readable, though let me know if you can think of a better name.

Reply:
We could also do something like `assert.is_not.open_state(infoview)`.

Reply:
I don't follow why that's not `.is_open`?

`.is_open` checks some internal state of course, but it's (public) API -- an API that should mean "is the window open", so I'd expect it to encapsulate all the implementation details of what it means to be open or not. If the window was ever not open but that function returned true that'd be a bug right? So I'd expect whatever `open_state` checks to be the same thing `is_open` checks internally.

But I see this is related to closed and open not being opposites (which I think is quite confusing even though this is math-land where topologists roam). I guess I can't really understand the way it's done here yet until I follow what the PR is meant for as a whole.

Reply:
There's a new state for the infoview - uninitialized. This check is the equivalent of:
```
assert.is_truthy(infoview.is_closed())
assert.is_falsy(infoview.is_open())
```

Let me know if you prefer it this way for readability.

Reply:
I prefer we make that not possible!

But I'm going to review this in a few hours.

But I would much prefer closed to be a superset of uninitialized, `is_closed` to just be `not is_opened`, and end-users never see uninitialized.

Reply:
Yeah it's actually not possible from a user's perspective, in that they'll never see an uninitialized infoview. So it's more of a sanity check for us.

Reply:
Also I don't think we can erase the distinction between closed and uninitialized - closed means explicitly closed by the user, so don't open. Uninitialized means open only if they've enabled auto-open.